packages/app/retainer/retainer.go

```diff
@@ -10,41 +10,56 @@ import (
 	"github.com/iotaledger/hive.go/core/syncutils"
 	"github.com/iotaledger/hive.go/core/workerpool"
 
+	"github.com/iotaledger/goshimmer/packages/core/commitment"
 	"github.com/iotaledger/goshimmer/packages/core/database"
 	"github.com/iotaledger/goshimmer/packages/core/epoch"
 	"github.com/iotaledger/goshimmer/packages/core/memstorage"
 	"github.com/iotaledger/goshimmer/packages/protocol"
 	"github.com/iotaledger/goshimmer/packages/protocol/congestioncontrol/icca/scheduler"
 	"github.com/iotaledger/goshimmer/packages/protocol/engine/consensus/blockgadget"
+	"github.com/iotaledger/goshimmer/packages/protocol/engine/notarization"
 	"github.com/iotaledger/goshimmer/packages/protocol/engine/tangle/blockdag"
 	"github.com/iotaledger/goshimmer/packages/protocol/engine/tangle/booker"
 	"github.com/iotaledger/goshimmer/packages/protocol/engine/tangle/virtualvoting"
 	"github.com/iotaledger/goshimmer/packages/protocol/ledger/utxo"
 	"github.com/iotaledger/goshimmer/packages/protocol/models"
 )
 
+const epochInCache = 20
+const (
+	prefixBlockMetadataStorage byte = iota
+
+	prefixCommitmentDetailsStorage
+)
+
 type Retainer struct {
-	workerPool     *workerpool.UnboundedWorkerPool
-	cachedMetadata *memstorage.EpochStorage[models.BlockID, *cachedMetadata]
-	blockStorage   *database.PersistentEpochStorage[models.BlockID, BlockMetadata, *models.BlockID, *BlockMetadata]
+	workerPool        *workerpool.UnboundedWorkerPool
+	cachedMetadata    *memstorage.EpochStorage[models.BlockID, *cachedMetadata]
+	blockStorage      *database.PersistentEpochStorage[models.BlockID, BlockMetadata, *models.BlockID, *BlockMetadata]
+	cachedCommitment  *memstorage.EpochStorage[commitment.ID, *cachedCommitment]
```

Comment:
Since commitments are not created too frequently and it's only 1 commitment per epoch I think a `cachedCommitment` is not necessary. We can probably simply create the `CommitmentDetails` directly and write them to disk. This simplifies the whole process massively, as there's only 1 storage, no eviction lock, no eviction and no separate cached commitment object